core-primitives/node-api/api-client-extensions/src/pallet_teerex.rs

```diff
@@ -17,43 +17,53 @@
 
 use crate::ApiResult;
 use itp_types::{Enclave, IpfsHash, ShardIdentifier};
-use sp_core::{Pair, H256 as Hash};
+use sp_core::Pair;
+use sp_rpc::number::NumberOrHex;
 use sp_runtime::MultiSignature;
-use substrate_api_client::{Api, ExtrinsicParams, RpcClient};
+use substrate_api_client::{Api, BalancesConfig, ExtrinsicParams, FromHexString, RpcClient};
+
+use codec::Decode;
+use core::str::FromStr;
 
 pub const TEEREX: &str = "Teerex";
 pub const SIDECHAIN: &str = "Sidechain";
 
 /// ApiClient extension that enables communication with the `teerex` pallet.
-pub trait PalletTeerexApi {
-	fn enclave(&self, index: u64, at_block: Option<Hash>) -> ApiResult<Option<Enclave>>;
-	fn enclave_count(&self, at_block: Option<Hash>) -> ApiResult<u64>;
-	fn all_enclaves(&self, at_block: Option<Hash>) -> ApiResult<Vec<Enclave>>;
+pub trait PalletTeerexApi<Runtime: BalancesConfig> {
+	fn enclave(&self, index: u64, at_block: Option<Runtime::Hash>) -> ApiResult<Option<Enclave>>;
+	fn enclave_count(&self, at_block: Option<Runtime::Hash>) -> ApiResult<u64>;
+	fn all_enclaves(&self, at_block: Option<Runtime::Hash>) -> ApiResult<Vec<Enclave>>;
 	fn worker_for_shard(
 		&self,
 		shard: &ShardIdentifier,
-		at_block: Option<Hash>,
+		at_block: Option<Runtime::Hash>,
 	) -> ApiResult<Option<Enclave>>;
 	fn latest_ipfs_hash(
 		&self,
 		shard: &ShardIdentifier,
-		at_block: Option<Hash>,
+		at_block: Option<Runtime::Hash>,
 	) -> ApiResult<Option<IpfsHash>>;
 }
 
-impl<P: Pair, Client: RpcClient, Params: ExtrinsicParams> PalletTeerexApi for Api<P, Client, Params>
+impl<P: Pair, Client: RpcClient, Params, Runtime> PalletTeerexApi<Runtime>
+	for Api<P, Client, Params, Runtime>
 where
 	MultiSignature: From<P::Signature>,
+	Params: ExtrinsicParams<Runtime::Index, Runtime::Hash>,
+	Runtime: BalancesConfig,
+	Runtime::Hash: FromHexString,
+	Runtime::Index: Into<u32> + Decode,
+	Runtime::Balance: TryFrom<NumberOrHex> + FromStr + Into<u128>,
```

Comment:
Hmmm, the trait bounds should be optimized in the api-client such that we don't have to redeclare them here. https://github.com/scs/substrate-api-client/issues/406

Reply:
That's a problem of the "outdated" branch 0.9.37. It's already been cleaned up drastically in the current master or tag v0.10.0. But going up there, would mean other necessary changes.